src/Adapter/Store/CommandHandler/AbstractStoreHandler.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Store\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Domain\AbstractObjectModelHandler;
+use PrestaShop\PrestaShop\Core\Domain\Store\Repository\StoreRepository;
+
+/**
+ * Class AbstractStoreHandler.
+ */
+class AbstractStoreHandler extends AbstractObjectModelHandler
```

Comment:
I wouldn't say its worth introducing abstract class just to reuse the constructor if it has no shared methods. 
Though i don't know, maybe its just me. Don't rush changing it before we get more opinions :man_shrugging:

Reply:
Ah, I don't agree here: all handlers use the StoreRepository instance. So if we don't use an abstract class, we'll have to copy/paste the argument in the services definitions... not good for me.